promql/parser/prettier_rules.md

```diff
@@ -0,0 +1,31 @@
+# Prettifying PromQL expressions
+This files contains rules for prettifying PromQL expressions.
+
+### Keywords
+`max_characters_per_line`: Maximum number of characters that will be allowed in a single stringified node
+
+## Rules
+1. A node exceeding the `max_characters_per_line` will qualify for split unless
+   1. It is a function call with one argument eg. `rate(foo[1m])`
+   2. It is a `VectorSelector`, `SubqueryExpr` and `MatrixSelector`
+2. Nodes that are nested within another node will be prettified only if they exceed the `max_characters_per_line`
+3. Following conditions do not involve splitting:
```

Comment:
Can you club this with point 1? Both are talking about when you do not split, and VectorSelector is repeated.